run commands on comments

Bullrich opened the pull request:
Added ability to parse comments when it's triggered by a `issue_comment` event.

Updated documentation showing how to execute the action's code.

Added composite actions to trigger the commands when passed to a matrix and push changes to the branch.

Resolves #1 and #2

Bullrich committed:
modified cmd name to not have whitespace

Bullrich committed:
added ability to parse commands

Bullrich committed:
added more verbosity

Bullrich committed:
added trimming of whitespace

Bullrich committed:
added composite command

So we don't need to have a very long action file

Bullrich committed:
added logic to push change

Bullrich committed:
moved output to index

Bullrich committed:
updated readme with new implementation

Bullrich committed:
self implemented action

Bullrich committed:
skip matrix when it's a PR

Bullrich committed:
added missing varialbe

We can remove it once the main branch gets updated

Bullrich committed:
fixed release check

Bullrich committed:
linted code

rzadp commented on .github/workflows/cmd-action.yml:
```diff
@@ -1,15 +1,45 @@
+    needs: [cmd-check]
+    if: ${{ github.event.issue.pull_request }}
+    continue-on-error: true
+    # We set the current machine here (to differ between ours and generic ones)
```
I don't get this comment, could you elaborate/rephrase?

Bullrich replied:
It‘s from the original proposal, so we can define custom self hosted machines for some more demanding jobs.

I will eventually add some more things here, but I would say to ignore it for now

rzadp commented on run/action.yml:
```diff
@@ -0,0 +1,41 @@
+      - name: Check changes
+        shell: bash
+        run: |
+          git config --global --add safe.directory /__w/${{ github.event.repository.name }}/${{ github.event.repository.name }}
```
Would it be cleaner to use `.` or `$(pwd)`?

Bullrich replied:
That‘s a good idea, I‘ll try that!

Bullrich replied:
I tried but end up reverting it as the running directory is `/home/runner/work/polkadot-sdk/polkadot-sdk`

rzadp replied:
Do you have a log of a failed run because of this? 🤔

Bullrich replied:
Here is tge test run: https://github.com/paritytech-stg/polkadot-sdk/actions/runs/8927469091/job/24520953847

rzadp replied:
And do you have a failed run log when you dont configure the safe directory at all?

Bullrich replied:
As I was having many issues, I decided to replace it with [stefanzweifel/git-auto-commit-action](https://github.com/stefanzweifel/git-auto-commit-action).

mutantcornholio commented on README.md:
```diff
@@ -10,27 +10,52 @@ Create a file named `.github/workflows/cmd-action.yml` and add the following:
+      - uses: paritytech/cmd-action/run@main
+        with:
+          branch: ${{ needs.cmd-check.outputs.branch }}
+          command: ${{ matrix.command }}
```
This sounds a bit extensive for the repository-side action. Sounds like something we will have to update frequently.  
Can we have this code inside of this repo, and just set the parameters?

Bullrich replied:
I‘m afraid I‘m not following you. Could you please rephrase it?

mutantcornholio replied:
I meant the whole action. There's logic in here, with checking out the PR branch, doing some calculations. I'd expect something more simple to be put in the repositories that use the action. 

We are already reusing `paritytech/cmd-action/run@main`, is it possible to move some of this logic in this repo, and not spread it outside?

Bullrich replied:
I now simplified it into it's own composite action.

Except the git push as that can be dependent on the docker image and I believe it's better to let the author customize it.

mutantcornholio commented on .github/workflows/cmd-action.yml:
```diff
@@ -1,15 +1,45 @@
           commands-directory: '.github/scripts'
           GITHUB_TOKEN: ${{ github.token }}
+      - name: Export branch name
+        run: echo "branch=$(git rev-parse --abbrev-ref HEAD)" >> "$GITHUB_OUTPUT"
```
Why do we need to parse branch here? Don't we have it in the variables?

Bullrich replied:
No, when it‘s generated from an issue comment we only have the PR number, but not the branch name, so I need to obtain it somehow. I could also do `gh pr view —json headRefName —jq „.headRefName“` to obtain the same result.

Bullrich committed:
updated action to use current dir

Bullrich committed:
added log

Bullrich committed:
reverted pwd directory

rzadp commented:
And do you have a failed run log when you dont configure this safe directory at all?On 2 May 2024, at 19:33, Javier Bullrich ***@***.***> wrote:﻿
@Bullrich commented on this pull request.



In run/action.yml:
> +        # Here it would get the script from previous step
+        run: |
+          echo "$COMMAND" >> command.sh
+          chmod +x command.sh
+          ./command.sh
+          rm command.sh
+        env:
+          COMMAND: ${{ inputs.command }}
+      - name: Report failure
+        if: ${{ failure() }}
+        shell: bash
+        run: echo "Command failed!"
+      - name: Check changes
+        shell: bash
+        run: |
+          git config --global --add safe.directory /__w/${{ github.event.repository.name }}/${{ github.event.repository.name }}

Here is tge test run: https://github.com/paritytech-stg/polkadot-sdk/actions/runs/8927469091/job/24520953847

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because your review was requested.Message ID: ***@***.***>

This communication and any attachments are confidential.

Bullrich committed:
added composite action to check

Bullrich committed:
added output object

Bullrich committed:
removed push changes from the action

If the user wants to have actions that push, lets make it so they manually add it

Bullrich committed:
applied simplified logic

Bullrich committed:
implemented action readme on itself

Bullrich committed:
removed irrelevant comment

Bullrich committed:
added validation for both images

rzadp commented on run/action.yml:
```diff
@@ -0,0 +1,30 @@
+      - name: Generate script
+        shell: bash
+        # Here it would get the script from previous step
+        run: |
```
I believe something like this would work in one line without the need for the file:

```
bash -c "$COMMAND"
```

Bullrich replied:
I worked flawlessly! Great recommendation!

Bullrich committed:
added bash command

Bullrich committed:
moved cmd-check to point to main
